lib/tasks/integration.rake

```diff
@@ -0,0 +1,96 @@
+INTEGRATION_TMP_DIR     = Rails.root.join('tmp', 'integration')
+INTEGRATION_FOREMAN_PID = INTEGRATION_TMP_DIR.join('server.pid')
+INTEGRATION_FOREMAN_LOG = Rails.root.join('log', 'integration.log')
+INTEGRATION_PROCFILE    = INTEGRATION_TMP_DIR.join('Procfile')
+RUN_SINGLE_WORKER_BIN   = Rails.root.join('lib', 'workers', 'bin', 'run_single_worker.rb')
+
+namespace :integration do
+  desc "Seed the database and configure assets for integration tests"
+  task :seed => [:env, "test:vmdb:setup", "evm:foreman:seed", :compile_assets]
+
+  desc "Setup the database for integration tests"
+  task :setup => [:db_setup, "evm:foreman:setup"] do
+    mkdir_p INTEGRATION_TMP_DIR
+    rm_f    INTEGRATION_PROCFILE
+    touch   INTEGRATION_PROCFILE
+  end
+
+  desc "Run a foreman server in the background for integration tests"
+  task :run_server => [:setup] do
+    pid = File.read(INTEGRATION_FOREMAN_PID).to_i if File.exist?(INTEGRATION_FOREMAN_PID)
+
+    if pid && (Process.getpgid(pid) rescue nil).present?
+      warn "foreman process already running"
+    else
+      foreman_cmd = %W[
+        foreman start --port=#{ENV['FOREMAN_PORT'] || '3000'}
+                      --root=#{Rails.root}
+                      --procfile=#{INTEGRATION_PROCFILE}
+      ].join(" ")
+
+      pid = Process.spawn(foreman_cmd, [:out, :err] => [INTEGRATION_FOREMAN_LOG, "a"])
+      Process.detach(pid)
+      File.write(INTEGRATION_FOREMAN_PID, pid)
+    end
+  end
+
+  desc "Stop server for integration tests"
+  task :stop_server do
+    if File.exist?(INTEGRATION_FOREMAN_PID)
+      Process.kill("TERM", File.read(INTEGRATION_FOREMAN_PID).to_i)
+      rm INTEGRATION_FOREMAN_PID
+    else
+      warn "PID file for server (#{INTEGRATION_FOREMAN_PID}) doesn't exist... moving on"
+    end
+  end
+
+  desc "With a UI worker for integration tests"
+  task :with_ui => :setup do
+    ui_config = "ui: env PORT=$PORT ruby #{RUN_SINGLE_WORKER_BIN} MiqUiWorker\n"
+    File.write(INTEGRATION_PROCFILE, ui_config, :mode => "a")
```

Comment:
Instead of generating this on the fly, can we just have a `Procfile.integration` file committed, and then in setup it's just copied to the right place?

Reply:
Also this appends to a file, but the file is empty to start with, so I'm not sure what this is doing?

Reply:
Yeah, probably.  My thoughts were that eventually we might need a generic worker at a minimum to test certain things in the app, and so I figured that being able to make this configurable for a local user might be ideal.

But, again, YAGNI and all that, so I am not opposed to just doing a `Procfile.integration` for now.

Reply:
"Sorta solved" this with the last couple of commits.  I am 50/50 on the last commit, and might still make a few tweaks to handling some of the file dependencies since I realize they might be better served as `file`/`dir` tasks instead of in a `.setup` method.

Regardless, it is still "generated" a bit because I want to make sure the `run_single_worker.rb` path is calculated and not hard coded, so using ERB for that bit at a minimum.